server/src/service.rs

```diff
@@ -83,9 +83,34 @@ impl Service {
     #[zbus(out_args("unlocked", "locked"))]
     pub async fn search_items(
         &self,
-        _attributes: HashMap<&str, &str>,
+        attributes: HashMap<String, String>,
     ) -> Result<(Vec<OwnedObjectPath>, Vec<OwnedObjectPath>), ServiceError> {
-        todo!()
+        let mut unlocked: Vec<OwnedObjectPath> = Vec::new();
+        let mut locked: Vec<OwnedObjectPath> = Vec::new();
+        let collections = self.collections.lock().await;
+
+        for collection in collections.iter() {
+            for item in collection.inner_items().await {
+                if item.attributes().await == attributes {
+                    if !item.is_locked().await {
```

Comment:
Flip the condition, no need for negation here.

Reply:
Done, thanks!